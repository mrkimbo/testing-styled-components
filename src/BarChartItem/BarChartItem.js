import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Bar, BarWrapper, Label, Legend, Total } from './BarChartItem.styles';

export default class BarChartItem extends PureComponent {
  render() {
    const { name, percentage } = this.props;
    return (
      <BarWrapper>
        <Legend>
          <Label>{name}</Label>
          <Total>{percentage}%</Total>
        </Legend>
        <Bar
          label={name}
          total={percentage}
        />
      </BarWrapper>
    );
  }
}

BarChartItem.propTypes = {
  name: PropTypes.string.isRequired,
  percentage: PropTypes.number.isRequired,
};
