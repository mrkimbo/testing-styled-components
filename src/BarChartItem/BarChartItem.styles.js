import styled from 'styled-components';

export const BarWrapper = styled.div`
  width: 100%;
  margin-bottom: 0.48em;
`;

export const Legend = styled.div`
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  justify-content: space-between;
  padding: 0.35em 0;
`;

export const Label = styled.label`
  color: Blue;
  transition: none;
  font-size: 16px;
  line-height: 1em;
`;

export const Total = styled.div`
  color: red;
  transition: none;
  font-size: 14px;
  line-height: 1em;
`;

export const Bar = styled.div`
  width: 100%;
  height: 3px;
  background-color: #CCC;
  position: relative;
  overflow: hidden;

  &:after {
    position: absolute;
    display: block;
    content: '';
    height: 3px;
    top: 0;
    left: 0;
    background-color: white;
  }
`;
