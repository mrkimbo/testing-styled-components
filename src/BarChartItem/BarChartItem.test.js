import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import BarChartItem from "./BarChartItem";
import { Bar, BarWrapper, Label, Legend, Total } from './BarChartItem.styles';

describe('<BarChartItem/>', () => {
  let props;
  const createComponent = () => shallow(<BarChartItem {...props}/>);
  
  beforeEach(() => {
    props = {
      name: 'Test Item',
      percentage: 0,
    };
  });
  
  it('should match snapshot', () => {
    expect(createComponent()).toMatchSnapshot();
  });
});

describe('[Styles]', () => {
  it('<Bar/> should match snapshot', () => {
    expect(shallow(<Bar/>)).toMatchSnapshot();
  });
  
  it('<BarWrapper/> should match snapshot', () => {
    expect(shallow(<BarWrapper/>)).toMatchSnapshot();
  });
  
  it('Label should match snapshot', () => {
    expect(shallow(<Label/>)).toMatchSnapshot();
  });
  
  it('<Legend/> should match snapshot', () => {
    expect(shallow(<Legend/>)).toMatchSnapshot();
  });
  
  it('<Total/> should match snapshot', () => {
    expect(shallow(<Total/>)).toMatchSnapshot();
  });
});
