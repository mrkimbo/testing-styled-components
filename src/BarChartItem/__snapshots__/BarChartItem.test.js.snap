// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarChartItem/> should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.label>
      Test Item
    </styled.label>
    <styled.div>
      0
      %
    </styled.div>
  </styled.div>
  <styled.div
    label="Test Item"
    total={0}
  />
</styled.div>
`;

exports[`[Styles] <Bar/> should match snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": true,
        "rules": Array [
          "
  width: 100%;
  height: 3px;
  background-color: #CCC;
  position: relative;
  overflow: hidden;

  &:after {
    position: absolute;
    display: block;
    content: '';
    height: 3px;
    top: 0;
    left: 0;
    background-color: white;
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`[Styles] <BarWrapper/> should match snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  width: 100%;
  margin-bottom: 0.48em;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`[Styles] <Legend/> should match snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  justify-content: space-between;
  padding: 0.35em 0;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`[Styles] <Total/> should match snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
  color: red;
  transition: none;
  font-size: 14px;
  line-height: 1em;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`[Styles] Label should match snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": true,
        "rules": Array [
          "
  color: Blue;
  transition: none;
  font-size: 16px;
  line-height: 1em;
",
        ],
      },
      "displayName": "styled.label",
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "label",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
